Support resource-defined authentication

landakram opened the pull request.

landakram committed:
Support resource-defined authentication

jessepollak commented on ripozo/decorators.py:
```diff
@@ -201,6 +204,22 @@ def wrapped(cls, request, *args, **kwargs):
 
 
+def assert_identity(cls, request):
+    identity_providers = getattr(cls, 'identity_providers', [])
```
It feels weird to have `authentication_required` and `identity_providers` be the two entrypoints into this API. Have you considered standardizing around either the "authentication" or "identity" language?

jessepollak reviewed (changes requested):
LGTM other than one small naming item.

landakram committed:
chore: standardize naming around identity

jessepollak reviewed (approved):
LGTM

landakram committed:
fix(identity): Stop after finding the first successful identity

landakram committed:
feat: Don't require identity

landakram committed:
Merge pull request #2 from clef/dont-require-identity

feat: Don't require identity

landakram committed:
fix(identity): Allow anonymous identities

Allow settings a parameter, allow_anonymous_identity, which specifies
that a resource can be accessed without an identity.

landakram committed:
Merge pull request #3 from clef/allow-anonymous-identity

fix(identity): Allow anonymous identities
